tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -919,39 +919,67 @@ void ROOT::Experimental::Internal::RPagePersistentSink::InitImpl(RNTupleModel &m
    fDescriptorBuilder.BeginHeaderExtension();
 }
 
-void ROOT::Experimental::Internal::RPagePersistentSink::InitFromDescriptor(const RNTupleDescriptor &descriptor)
+std::unique_ptr<ROOT::Experimental::RNTupleModel>
+ROOT::Experimental::Internal::RPagePersistentSink::InitFromDescriptor(const RNTupleDescriptor &srcDescriptor)
 {
-   {
-      auto model = descriptor.CreateModel();
-      Init(*model.get());
-   }
+   // Create new descriptor
+   fDescriptorBuilder.SetSchemaFromExisting(srcDescriptor);
+   const auto &descriptor = fDescriptorBuilder.GetDescriptor();
 
-   auto clusterId = descriptor.FindClusterId(0, 0);
+   // Create column/page ranges
+   const auto nColumns = descriptor.GetNPhysicalColumns();
+   fOpenColumnRanges.reserve(fOpenColumnRanges.size() + nColumns);
+   fOpenPageRanges.reserve(fOpenPageRanges.size() + nColumns);
+   for (DescriptorId_t i = 0; i < nColumns; ++i) {
+      const auto &column = descriptor.GetColumnDescriptor(i);
+      RClusterDescriptor::RColumnRange columnRange;
+      columnRange.fPhysicalColumnId = i;
+      columnRange.fFirstElementIndex = column.GetFirstElementIndex();
+      columnRange.fNElements = 0;
+      columnRange.fCompressionSettings = GetWriteOptions().GetCompression();
+      fOpenColumnRanges.emplace_back(columnRange);
+      RClusterDescriptor::RPageRange pageRange;
+      pageRange.fPhysicalColumnId = i;
+      fOpenPageRanges.emplace_back(std::move(pageRange));
```

Comment:
Is there a reason for moving one but not the other? I think both structures aren't big nor are we on a performance critical path, so even just `push_back` should be fine

Reply:
`RPageRange` has a deleted destructor so it must be moved (see also line 876)